core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts

```diff
@@ -577,31 +589,41 @@ export class EditEmaEditorComponent implements OnInit, OnDestroy {
                 });
             },
             [NG_CUSTOM_EVENTS.SAVE_PAGE]: () => {
-                if (payload) {
-                    const { pageContainers, didInsert } = insertContentletInContainer({
-                        ...payload,
-                        newContentletId: detail.payload.contentletIdentifier
-                    });
+                const { shouldReloadPage, contentletIdentifier } = detail.payload;
 
-                    if (!didInsert) {
-                        this.handleDuplicatedContentlet();
+                if (shouldReloadPage) {
+                    this.store.reload(this.queryParams);
 
-                        return;
-                    }
+                    return;
+                }
 
-                    // Save when created
-                    this.store.savePage({
-                        pageContainers,
-                        pageId: payload.pageId,
-                        params: this.queryParams,
-                        whenSaved: () => {
-                            this.dialog.resetDialog();
-                            this.reloadIframe();
-                        }
-                    });
-                } else {
+                if (!payload) {
                     this.reloadIframe(); // We still need to reload the iframe because the contentlet is not in the container yet
+
+                    return;
+                }
+
```

Comment:
So let me see if I get this:

First we reload the store

Then we reload the iframe to get the new data.

I know store and iframe are independent a little bit, but can this be a race condition?

Reply:
@fmontes each `if` statement has its `return`. There shouldn't be a race condition

Reply:
Done!